packages/kamino-sdk/src/Kamino.ts

```diff
@@ -249,6 +250,34 @@ export class Kamino {
     }
   }
 
+  /**
+   * Get the strategies share data (price + balances) of the Kamino whirlpool strategies that match the filters
+   * @param strategy
+   */
+  async getStrategyShareDataForStrategies(strategyFilters: StrategiesFilters): Promise<Array<ShareDataWithAddress>> {
+    let result: Array<ShareDataWithAddress> = [];
+    let strategiesWithAddresses = await this.getAllStrategiesWithFilters(strategyFilters);
+    for (let strategyState of strategiesWithAddresses) {
+      const sharesFactor = Decimal.pow(10, strategyState.strategy.sharesMintDecimals.toString());
```

Comment:
All of the logic inside the for loop can be replaced by calling getStrategyShareData and pushing that to the result array